### PR TITLE
Using redux in react components

### PR DESCRIPTION
**Pull request by Sboursen**
# Description

This is the 3rd milestone in the BookStore project.

## Milestone requirements

- [x] Used react-redux to implement the add book functionality
- [x] Used react-redux to implement the remove book functionality
- [x] Used react-redux to implement the check status functionality

## General requirements

- [x] There are no linter errors.
- [x] Followed the **Gitflow**.
- [x] The documentation is professional.

## JavaScript requirements

- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/components/book-card/book-card.js [hunk 1]
```diff
@@ -5,11 +5,12 @@ import BookProgress from '../book-progress/book-progress';
 
 export default function BookCard(props) {
   const {
-    percent, chapter, genre, title, author,
+    id, percent, chapter, genre, title, author,
```
Comment (micheaol) [inline src/components/book-card/book-card.js:8]: Please ensure that your `list of books` displays books from `Redux store` (by using `useSelector` function) according to the project requirement.

image

Please check [HERE for more details](https://react-redux.js.org/api/hooks).

File: src/components/form/from.js [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
 export default function Form() {
   const [bookTitle, setBookTitle] = useState('');
   const [selected, setSelected] = useState('');
+  const [valid, setValid] = useState([true, true]);
+  const dispatch = useDispatch();
+
   const handleChange = (e) => {
     setBookTitle(e.target.value);
   };
+
```
Comment (micheaol) [inline src/components/form/from.js]: Please ensure that your `New Book React Component` has a local state with `title`, `author` according to the project requirement.

image

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @Sboursen , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/form/from.js (hunk 2 above)
Comment (Sboursen) [reply src/components/form/from.js]: I don't understand where is the input relative to 'author'. 
We only have an input to 'title' and another one for categories li it is shown in the reference design [image](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/images/bookstore.png?raw=true).

In addition, the project requirements that I have are different from the ones that you took a screenshot of: [link](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md)

**Comment by Sboursen** [comment]
Hi @micheaol 

Thank you for the feedback.
I am a little bit confused about the changes that you requested.
I left you some comments about what I don't understand above :arrow_up:

File: src/components/book-card/book-card.js (hunk 1 above)
Comment (Sboursen) [reply src/components/book-card/book-card.js:8]: The `List of books` is being generated using the `Redux store`

Please see the following [books.js](https://github.com/Sboursen/BookstoreCMS/blob/using-redux-in-react-components/src/routes/books/books.js) page.

File: src/components/form/from.js (hunk 2 above)
Comment (micheaol) [reply src/components/form/from.js]: @Sboursen ,

I believe this was a requirement in a previous milestone:

image

Therefore, I highly suggest you implement the author's input, as you will need it, and will make your life easier on the next milestones.

### Explanation
The design that you can see in the sneak peek, is just for the styling part of the project.

### Hints
You could shorten the "Book title" input, and in that space insert your "Author" input, next to it, and you will preserve your styling. Like this:

image

I hope this helps.

Please find below your project requirement from the link provided by you above:

image


>

Comment (Sboursen) [reply src/components/form/from.js]: Thank you for the clarification :pray:

Comment (micheaol) [reply src/components/form/from.js]: @Sboursen 
I am glad, I am able to help. Happy coding!

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Sboursen,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**